libs/libcommon/include/common/fshorten.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+// Based on https://www.experts-exchange.com/questions/26210801/ftruncate-the-file-in-front.html
```

Comment:
It's unclear what is the license of this code...
Nevertheless, this way is dangerous (it can lead to data corruption easily).
Creating temporary file, filling it and renaming is more safe.